### PR TITLE
Refactored Spanish fiscalization tests and added tests to support Zero VAT Invoices

### PR DESCRIPTION
**Pull request by inakiesteve**
## Description

Minor refactor of the Spanish fiscalization tests + adding some tests to show how Invoices with Zero VAT should be handled in the Spanish fiscalization.

Fixes #issue

## Type of change
- [ ] Bug fix.
- [ ] Feature.
- [ ] Consolidation.

## Checklist
- [ ] Is breaking change.
- [ ] Documentation updated.
- [x ] Tests included (please specify cases).

### CODE REVIEW

File: src/Spain/Mews.Fiscalizations.Spain.Tests/Basics.cs [hunk 1]
```diff
@@ -10,25 +10,43 @@ namespace Mews.Fiscalizations.Spain.Tests.IssuedInvoices;
...
     ).Success.Get();
 
+    private static readonly TaxExemptItem[] UntaxedItems =
+    [
+        new(Amount.Create(20m).Success.Get(), CauseOfExemption.OtherGrounds)
+    ];
+
+    private static readonly TaxRateSummary[] TaxedItems =
+    [
+        GetTaxRateSummary(21m, 42.07M)
```
Comment (abdallahbeshi) [inline src/Spain/Mews.Fiscalizations.Spain.Tests/Basics.cs:38]: nitpick: naming the params should make it more clear what these numbers mean without diving into the code :)

File: src/Spain/Mews.Fiscalizations.Spain.Tests/Basics.cs [hunk 2]
```diff
@@ -10,25 +10,43 @@ namespace Mews.Fiscalizations.Spain.Tests.IssuedInvoices;
...
+    ];
+
+    private static readonly TaxRateSummary[] InvalidTaxedItems =
+    [
+        GetTaxRateSummary(0, 42.07M)
```
Comment (abdallahbeshi) [inline src/Spain/Mews.Fiscalizations.Spain.Tests/Basics.cs:43]: Same here

File: src/Spain/Mews.Fiscalizations.Spain.Tests/Basics.cs [hunk 3]
```diff
@@ -56,12 +74,26 @@ public async Task CheckNif()
     [Retry(RetryCount)]
     public async Task PostInvoice()
```
Comment (abdallahbeshi) [inline src/Spain/Mews.Fiscalizations.Spain.Tests/Basics.cs:75]: I know this is an old test, but how about we improve its naming? I am thinking something like "PostInvoice_WithValidData_Succeeds"

File: src/Spain/Mews.Fiscalizations.Spain.Tests/Basics.cs [hunk 4]
```diff
@@ -56,12 +74,26 @@ public async Task CheckNif()
...
     {
-        await SuccessfullyPostInvoice(Client);
+        await SuccessfullyPostInvoice(Client, GetInvoice(issuer: Issuer, taxRateSummaries: TaxedItems, taxExemptItems: UntaxedItems));
+    }
+
+    [Test]
+    [Retry(RetryCount)]
+    public async Task PostZeroVatInvoice()
```
Comment (abdallahbeshi) [inline src/Spain/Mews.Fiscalizations.Spain.Tests/Basics.cs:82]: PostZeroVatInvoice_WithValidData_Succeeds

File: src/Spain/Mews.Fiscalizations.Spain.Tests/Basics.cs [hunk 5]
```diff
@@ -72,8 +104,20 @@ private async Task<SimplifiedInvoice> SuccessfullyPostInvoice(Client client)
...
         var errorMessage = String.Join(System.Environment.NewLine, responseErrorMessages);
         Assert.That(response.Success.Get().Result, Is.EqualTo(RegisterResult.Correct), errorMessage);
+    }
 
-        return invoice;
+    private async Task UnsuccessfullyPostInvoice(Client client, SimplifiedInvoice invoice)
+    {
+        var model = SimplifiedInvoicesToSubmit.Create(
+            header: new Header(Issuer, CommunicationType.Registration),
+            invoices: new[] { invoice }
```
Comment (abdallahbeshi) [inline src/Spain/Mews.Fiscalizations.Spain.Tests/Basics.cs:113]:
```suggestion
            invoices: [invoice]
```

File: src/Spain/Mews.Fiscalizations.Spain.Tests/Basics.cs [hunk 6]
```diff
@@ -72,8 +104,20 @@ private async Task<SimplifiedInvoice> SuccessfullyPostInvoice(Client client)
...
+        ).Success.Get();
+
+        var response = await client.SubmitSimplifiedInvoiceAsync(model);
+
+        var responseErrorMessages = response.Success.Get().Invoices.Select(i => i.ErrorMessage).Flatten();
+        var errorMessage = String.Join(System.Environment.NewLine, responseErrorMessages);
+        Assert.That(response.Success.Get().Result, Is.EqualTo(RegisterResult.AllIncorrect), errorMessage);
```
Comment (abdallahbeshi) [inline src/Spain/Mews.Fiscalizations.Spain.Tests/Basics.cs:120]: Duplicate code, you can abstract the logic to a method that accepts the expected result

**Review by abdallahbeshi — APPROVED** [review]
